### DISCUSSION TITLE
Video Scripting

### DISCUSSION THREAD

**Discussion by parsoli** [JVital2013/vitality-goes#33] [Q&A] [answered]
Hello there.  Excellent tool, thanks for your hard work.  A couple of questions:

I'm starting to play with scripting.  Note the below during the CreateVideos-ABI.sh run.  It's injecting a double //.  I can't find in the script config a double //.  Curious as to your thoughts.

Once the videos are created and output to /videos, did you have any plans on how to represent the collection?  Right now browsing to that renders a blank page.  Maybe I have missed something in the instructions?

Thanks!

[2022-12-21 14:31:06] Creating GOES16Ch13Enhanced_M2...
find: ‘/var/www/html/NASLink//goes16/fd/sanchez’: No such file or directory
[2022-12-21 14:31:32] No images found for GOES16Sanchez; skipping
find: ‘/var/www/html/NASLink//composite’: No such file or directory
[2022-12-21 14:31:32] No images found for Composite; skipping

**Comment by parsoli** [comment]
As I review more logs.  I notice that may be normal and it's reporting this because I do not have a folder for Sanchez or Composite.  So answered my own question there.  Curious on HTML to render the video's in a format you have for the images however.

**Reply by parsoli** [reply]
When running the Sanchez.sh script, I'm receiving the below.

[2022-12-21 16:17:04] Creating GOES16_FD_sanchez_20221120T180020Z.jpg...
**./Sanchez.sh: line 105: -q: command not found**

From the Sanchez.sh script I see the below block.  The last line is what is referenced as Line 105.  It's stating 'command not found'.  Do we know what this -q operator is supposed to be doing?

#Build Underlay
		rm /tmp/underlay.jpg > /dev/null 2>&1
		satelliteMonth=$(date -u -d "$thisDate ${thisTime:0:2}:${thisTime:2:2}:${thisTime:4:2} + 5 hour" +"%m")
		xplanet -config "$(dirname "$(readlink -fm "$0")")/Resources/xplanet-$satelliteMonth" -body earth -projection rectangular -num_times 1 -geometry 21600x10800 -date $thisDate.$thisTime -output /tmp/underlay.jpg
		**$sanchezPath -q -s $srcImg -u /tmp/underlay.jpg -r $((10850/$(identify -format '%w' $srcImg))) -n -o $sanchezDstPath16/$newName**

**Reply by JVital2013** [reply]
-q tells Sanchez to run silently. If it says "-q command not found", that means your sanchezPath variable is not set in scriptconfig.ini (and it's trying to execute the next string, which is -q - and that makes no sense to Bash)

Double-check that you have sanchezPath in scriptconfig.ini set, and that there are no spaces around the = sign

**Reply by parsoli** [reply]
Thank you.  I REM'd out lines I do not have images for.  Re-Ran Sanchez.sh and now receive:
**./Sanchez.sh: line 105: /home/sparsons/vitality-goes/scripts/sanchez-v1.0.22-linux-arm/Sanchez: cannot execute binary file: Exec format error**

sanchezSrcPath16=/var/www/html/NASLink/goes16/fd/ch13
#sanchezSrcPath17=/var/www/html/NASLink/goes17/fd/ch13
#sanchezSrcPath18=/var/www/html/NASLink/goes18/fd/ch13
sanchezDstPath16=/var/www/html/NASLink/goes16/fd/sanchez
#sanchezDstPath17=/var/www/html/NASLink/goes17/fd/sanchez
#sanchezDstPath18=/var/www/html/NASLink/goes18/fd/sanchez
#dstPathComposite=/var/www/html/NASLink/composite
sanchezPath=/home/sparsons/vitality-goes/scripts/sanchez-v1.0.22-linux-arm/Sanchez

**Reply by JVital2013** [reply]
"Exec format error" means it isn't understanding the SANCHEZ binary. This can either be because it's corrupted, or you downloaded it for the wrong architecture. 

Looks like you downloaded the ARM version, which would be correct for a raspberry pi or the like, but incorrect for a typical x86-64 PC. What type of system do you have?

**Reply by parsoli** [reply]
Another boneheaded move on my part.  Grabbed ARM instead of 64.  Sanchez is muy bueno now.  :)

**Comment by JVital2013** [comment, marked as answer]
Videos should to be kept in DOCUMENT_ROOT/videos, which is usually /var/www/html/videos on Linux.

After they're in there, take a look at [https://github.com/JVital2013/vitality-goes/blob/main/docs/config.md#abiini-mesoini-and-l2ini](https://github.com/JVital2013/vitality-goes/blob/main/docs/config.md#abiini-mesoini-and-l2ini).

Each image defined in abi.ini, l2.ini, meso.ini, and emwin.ini can have a `videoPath` attribute set. Match this to the video name in your videos folder, and a "Timelapse" option will appear in the web interface/

![image](https://user-images.githubusercontent.com/24253715/208962488-7c2e1839-5cd9-4a3d-bacd-935be30c6581.png)
![image](https://user-images.githubusercontent.com/24253715/208963171-a1cea5e8-05b0-4a3b-913d-fcd2c0a92fe1.png)
![image](https://user-images.githubusercontent.com/24253715/208963325-2fd61dce-19ab-4e2d-9c28-d57432dcf9e7.png)

**Reply by parsoli** [reply]
Ok, was able to get Timelapse to show up.  However, the video won't start.  I get a spinning circle.  If I right click and say Copy URL, I get the below.  Nope the path injected
**https://goes.elburnweather.org/videos//var/www/html/videos/GOES16Ch2.mp4**
I had put the videopath = /var/www, etc...

So I changed the videopath to the below:

[fdch02_16]
path = {GOES16}/goes16/fd/ch02/
title = "GOES 16 - Channel 2 (Red)"
videoPath = https://goes.elburnweather.org/videos/GOES16Ch2.mp4

Similar result in that the path is invalid.
https://goes.elburnweather.org/videos/https://goes.elburnweather.org/videos/GOES16Sanchez.mp4

So is the videopath = the linux file system path, the HTTPS: path or just the title as shown in your document?  I've tried all three ways, so I'm assuming there is something I'm missing.  Thanks again!

**Reply by JVital2013** [reply]
Glad to hear you got it! I assume your solution was to remove `/var/www/html/videos/` from the video path, and just have `GOES16Ch2.mp4`?

**Reply by parsoli** [reply]
Yup!  All working now Pal.  Thank you!
